### ISSUE TITLE
gyp: Call to 'pg_config --libdir' returned exit status 127. while trying to load binding.gyp

### ISSUE THREAD

**Issue by groucho-zz** [nodejs/node-gyp#223]
> node-gyp rebuild || (exit 0)

/bin/sh: 1: pg_config: not found
gyp: Call to 'pg_config --libdir' returned exit status 127. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:416:16)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:754:12)
gyp ERR! System Linux 3.2.0-39-generic
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/geddy/node_modules/model/node_modules/pg
gyp ERR! node -v v0.10.0
gyp ERR! node-gyp -v v0.8.5
gyp ERR! not ok 

encountered error while trying to install geddy on node v0.10.0

**Comment by kdemarest** [comment]
Sorry to be dense, but I see a lot of people confused by this. Can you please confirm that when I see this:
```
gyp: Call to 'pg_config --libdir' returned exit status 127. while trying to load binding.gyp
```
What is REALLY happening is that the people who made the module are using node-gyp to do their install, and it is THEIR problem. Not node-gyp's problem at all, right?  

When it follows up with:
```
gyp ERR! stack Error: `gyp` failed with exit code: 1
```
I get this really negative impression of node-gyp because every time I have an `npm install` issue, the failure is clearly gyp's problem!  You might put something in bold on the github readme way up top, for the idiots like me... Or perhaps change how errors are reported just a bit.

**Comment by leafoflegend** [comment]
I want to throw this in here as this happens a lot for me.

A: Don't use two different db libraries by accident: i.e. `pg` and `pg.js`.
B: Be cautious using things like sequelize that you may have flagged the 'native' option in initializing your connection to the DB. This will also often cause this error.
C: Lastly, I often remove the `pg-native` library entirely. Causes strange errors that are not consistent with node-gyp.

I know it's not entirely on topic - but this is the first google search result for this problem and its a common one. Just want to help whoever ends up here.

**Comment by RahmatAliMalik5** [comment]
If you have encountered this `gyp` error `Call to 'pg_config --libdir'` or something like that, and you have installed `sequelize` and `pg` with `pg-native`, try to run `npm i` with `sudo`:

```
sudo npm i
```

But first confirm by running `pg_config` that `pg_config` command is working and returning some values as well.

In my case, I have previously installed `pg-native` and it's requirements (can be seen on `pg-native` npm page [here](https://www.npmjs.com/package/pg-native)) with `sequelize`. The project was working fine. Then, one of my employee added some new package and pushed the code. I pulled and have to run `npm i` which was going failed with `Call to 'pg_config*` error. `sudo npm i`  worked for me.

I think `gyp` tries to install some package which requires to run `pg_config`. It goes failed as it does not have proper permissions to run `pg_config`.

I added it here so anyone comes here in search of a solution regarding `seqeulize` and `pg-native` with this error can try it.
